From patchwork Fri Jan  4 19:57:14 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Laura Abbott <labbott@redhat.com>
X-Patchwork-Id: 10748803
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 75D7314E2
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri,  4 Jan 2019 19:57:37 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 6636F285EA
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri,  4 Jan 2019 19:57:37 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 597D1285F2; Fri,  4 Jan 2019 19:57:37 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id F2DE6285EA
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri,  4 Jan 2019 19:57:36 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726036AbfADT5T (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Fri, 4 Jan 2019 14:57:19 -0500
Received: from mail-qk1-f196.google.com ([209.85.222.196]:34451 "EHLO
        mail-qk1-f196.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1725958AbfADT5T (ORCPT <rfc822;kvm@vger.kernel.org>);
        Fri, 4 Jan 2019 14:57:19 -0500
Received: by mail-qk1-f196.google.com with SMTP id q8so22203478qke.1
        for <kvm@vger.kernel.org>; Fri, 04 Jan 2019 11:57:19 -0800 (PST)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=nu73my8EyoywVEqped8htPAV7wuMwc2otKpVbNgVKi8=;
        b=JldVfLvH5c84O009nm7zLoeqm2svmtmDj34NgXRjDCi6e7/Rk15A+1I7/eBvjtjDfn
         x2UMlMiBctYZbQL40diunBv4ma6HSUxJ3cU+dqKa2D9eIadT2IUG8jhwVWt4jwBMtp9J
         ZioAZxTVUB5rGR7exZzziWsttAutGrEI6PIG/UXiuTWWstGbUSa7fQ2tABw2cL//HwZt
         TZ2KwNC/aSfQjgM8gylSut2lctQotk8pwYF5bA9TwyNlfRFmXNQHR9JcNezAT31UOO+e
         GSv7MkaLyG9xJ4M8e664Icwr1HTacEl8xzsm1BV+Yf/5DNieb78m7sqc3vM913sHc3Zp
         +sDg==
X-Gm-Message-State: AJcUukcdoJiE+sp7llIcxdL0Eut2Q80rB5OEpChDxxHIvkDQ/yA83G31
        6+3J7GCoLP+qEY3MNAM6i1OlLw==
X-Google-Smtp-Source: 
 ALg8bN5P/1jjRM2lDhsRg4H6VUN3kA7+IjuknOjwkecgMojiSefjQ0Akib1ZEXt+274T8fS+9ABqeQ==
X-Received: by 2002:a37:5a05:: with SMTP id o5mr48160152qkb.126.1546631838774;
        Fri, 04 Jan 2019 11:57:18 -0800 (PST)
Received: from labbott-redhat.redhat.com ([2601:602:9800:dae6::814b])
        by smtp.gmail.com with ESMTPSA id
 p47sm38210529qta.36.2019.01.04.11.57.17
        (version=TLS1_2 cipher=ECDHE-RSA-CHACHA20-POLY1305 bits=256/256);
        Fri, 04 Jan 2019 11:57:17 -0800 (PST)
From: Laura Abbott <labbott@redhat.com>
To: Alexey Kardashevskiy <aik@ozlabs.ru>,
        Alex Williamson <alex.williamson@redhat.com>,
        Michael Ellerman <mpe@ellerman.id.au>
Cc: Laura Abbott <labbott@redhat.com>, kvm@vger.kernel.org,
        Linux Kernel Mailing List <linux-kernel@vger.kernel.org>
Subject: [PATCH] vfio_pci: Add local source directory as include
Date: Fri,  4 Jan 2019 11:57:14 -0800
Message-Id: <20190104195714.30045-1-labbott@redhat.com>
X-Mailer: git-send-email 2.20.1
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Commit 7f92891778df ("vfio_pci: Add NVIDIA GV100GL [Tesla V100 SXM2]
subdriver") introduced a trace.h file in the local directory but
missed adding the local include path, resulting in compilation
failures with tracepoints:

In file included from drivers/vfio/pci/trace.h:102,
                 from drivers/vfio/pci/vfio_pci_nvlink2.c:29:
./include/trace/define_trace.h:89:42: fatal error: ./trace.h: No such file or directory
 #include TRACE_INCLUDE(TRACE_INCLUDE_FILE)

Fix this by adjusting the include path.

Fixes: 7f92891778df ("vfio_pci: Add NVIDIA GV100GL [Tesla V100 SXM2] subdriver")
Signed-off-by: Laura Abbott <labbott@redhat.com>
Reviewed-by: Alexey Kardashevskiy <aik@ozlabs.ru>
Reviewed-by: Cornelia Huck <cohuck@redhat.com>
---
I'd still like to echo my sentiment that this should not be a def_bool.
We hit this error on our internal testing and we couldn't even turn
off the driver until we fixed this.
---
 drivers/vfio/pci/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/vfio/pci/Makefile b/drivers/vfio/pci/Makefile
index 9662c063a6b1..08d4676a8495 100644
--- a/drivers/vfio/pci/Makefile
+++ b/drivers/vfio/pci/Makefile
@@ -1,3 +1,4 @@
+ccflags-y                               += -I$(src)
 
 vfio-pci-y := vfio_pci.o vfio_pci_intrs.o vfio_pci_rdwr.o vfio_pci_config.o
 vfio-pci-$(CONFIG_VFIO_PCI_IGD) += vfio_pci_igd.o
